### PR TITLE
[Issue #5] Database - drugs table display

### PR DESCRIPTION
**Pull request by kozych**
Issue #5: Database - drugs table display

added DrugActivity.kt, activity_drug.xml
added drug_item.xml
added custom adapter - DrugAdapter.kt
enabled viewBinding in build.gradle
added button to MainActivity.kt which starts DrugActivity.kt (this is only for testing purposes)
added function getAllDrugs() to DrugDAO.kt
added val allDrugs to DrugRepository.kt
added val allDrugs to DrugViewModel.kt
optimized imports and rearranged code in MainActivity.kt
now MainActivity.kt is using viewBinding

### CODE REVIEW

**Comment by michalmobi** [comment]
@kozych please add reviewers

**Comment by michalmobi** [comment]
@kozych please add reviewers

File: app/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -1,3 +1,14 @@
...
+    <string name="drug_fragment_title">Add drug to database</string>
+    <string name="tv_drug_name">Drug name</string>
+    <string name="hint_drug_name">Drug name</string>
+    <string name="tv_dose">Dose</string>
+    <string name="hint_dose">Dose</string>
+    <string name="tv_dose_date">Time of first dose</string>
+    <string name="hint_select_date">Select date</string>
+    <string name="tv_dose_interval">Dose interval</string>
+    <string name="hint_dose_interval">Take drug every x hours</string>
+    <string name="submit">Submit</string>
```
Comment (AlicjaKowalska) [inline app/src/main/res/values/strings.xml:13]: There's only needed "tv_dose" and "tv_dose_interval" here or even there's no need to use those strings at all if android:text will be removed from the TextViews

Comment (kozych) [reply app/src/main/res/values/strings.xml]: You are right, removed unused strings

**Review by AlicjaKowalska — APPROVED** [review]

File: app/src/main/res/layout/drug_item.xml [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+        <View
+            android:id="@+id/divider"
+            android:layout_width="match_parent"
+            android:layout_height="1dp"
+            android:layout_marginTop="8dp"
+            android:background="?android:attr/listDivider" />
+
+    </LinearLayout>
+
+</androidx.cardview.widget.CardView>
```
Comment (AlicjaKowalska) [inline app/src/main/res/layout/drug_item.xml:64]: Please stick with the design we agreed on

Comment (kozych) [reply app/src/main/res/layout/drug_item.xml]: Please check if you agree with this design for now 
image

Comment (AlicjaKowalska) [reply app/src/main/res/layout/drug_item.xml]: Looks good 👍

**Comment by piotrgos45** [comment]
I think that dose could be floating-point number. Sometimes you take half a tablet. But generally looks good for me!

File: app/src/main/java/com/example/patientassistant/Adapters/DrugAdapter.kt [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+        return DrugViewHolder(binding)
+    }
+
+    override fun onBindViewHolder(holder: DrugViewHolder, position: Int) {
+        val currentDrug: Drug = drugs[position]
+        val drugName = currentDrug.name
+        val drugSupportText = buildString {
+            append(currentDrug.dose)
+            append(" ")
+            if (currentDrug.dose > 1) append("tablets ") else append("tablet ")
```
Comment (michalmobi) [inline app/src/main/java/com/example/patientassistant/Adapters/DrugAdapter.kt:26]: use strings.xml for this

File: app/src/main/java/com/example/patientassistant/Adapters/DrugAdapter.kt [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+            append("every ")
+            if (currentDrug.interval > 1) {
+                append(currentDrug.interval)
+                append(" hours")
+            } else {
+                append("hour")
+            }
+        }
+        with(holder) {
```
Comment (michalmobi) [inline app/src/main/java/com/example/patientassistant/Adapters/DrugAdapter.kt:35]: can we do `with(holder.binding)`?

File: app/src/main/java/com/example/patientassistant/Adapters/DrugAdapter.kt [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+            with(binding) {
+                tvDrugCapital.text = drugName.uppercase()[0].toString()
+                tvDrugName.text = drugName
+                tvDrugSupportText.text = drugSupportText
+            }
+        }
+    }
+
+    override fun getItemCount(): Int {
```
Comment (michalmobi) [inline app/src/main/java/com/example/patientassistant/Adapters/DrugAdapter.kt:44]: `override fun getItemCount() = drugs.size`

File: app/src/main/java/com/example/patientassistant/Fragment/DrugListFragment.kt [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+        drugActivity = activity as DrugActivity
+
+        val drugAdapter = DrugAdapter()
+        binding.recyclerViewDrugs.layoutManager = LinearLayoutManager(drugActivity)
+        binding.recyclerViewDrugs.adapter = drugAdapter
+
+        drugActivity.drugViewModel.allDrugs.observe(drugActivity) { drugs ->
+            drugAdapter.setDrug(drugs)
+        }
+
```
Comment (michalmobi) [inline app/src/main/java/com/example/patientassistant/Fragment/DrugListFragment.kt:40]: unnecessary new line

**Review by michalmobi — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/example/patientassistant/Adapters/DrugAdapter.kt (hunk 3 above)
Comment (kozych) [reply app/src/main/java/com/example/patientassistant/Adapters/DrugAdapter.kt]: done

File: app/src/main/java/com/example/patientassistant/Adapters/DrugAdapter.kt (hunk 4 above)
Comment (kozych) [reply app/src/main/java/com/example/patientassistant/Adapters/DrugAdapter.kt]: right, done

File: app/src/main/java/com/example/patientassistant/Adapters/DrugAdapter.kt (hunk 5 above)
Comment (kozych) [reply app/src/main/java/com/example/patientassistant/Adapters/DrugAdapter.kt]: done

File: app/src/main/java/com/example/patientassistant/Fragment/DrugListFragment.kt (hunk 6 above)
Comment (kozych) [reply app/src/main/java/com/example/patientassistant/Fragment/DrugListFragment.kt:40]: removed this line

**Review by michalmobi — APPROVED** [review]

**Pull request merged**